src/dftbp/dftb/periodic.F90

```diff
@@ -94,6 +95,36 @@ module dftbp_dftb_periodic
 
   end type TNeighbourList
 
+
+  !> Contains neighbour list instance and symmetry specific entries
+  type TSymNeighbourList
+
+    !> Neighbour list instance
+    type(TNeighbourList), allocatable :: neighbourList
+
+    !> Number of all interacting atoms, including periodic images
+    integer :: nAllAtom
+
+    !> Coordinates of all interacting atoms, including periodic images
+    real(dp), allocatable :: coord(:,:)
+
+    !> Species of all interacting atoms, including periodic images
+    integer, allocatable :: species(:)
+
+    !> Mapping of all atoms onto atoms in the central cell
+    integer, allocatable :: img2CentCell(:)
+
+    !> Shift vector index for every interacting atom, including periodic images
+    integer, allocatable :: iCellVec(:)
+
+    !> Sparse array indexing for the start of atomic blocks in data structures
+    integer, allocatable :: iPair(:,:)
+
+    !> Total number of elements in a sparse structure (ignoring extra indices like spin)
+    integer :: sparseSize
```

Comment:
Does this really belong in this data structure? It is also related to basis set size, not purely geometry.

Reply:
No, I think I "temporarily" added it to the sym-neighlist type in order to avoid overwriting the usual `this%sparseSize` variable and forgot to search for a more appropriate place afterwards. What about introducing `this%sparseSizeSym` or something similar?

Reply:
The cleaner thing might be a type for sparse structure descriptors (we have something sort of like this for the dense case).